app/assets/javascripts/discourse/app/mixins/password-validation.js

```diff
@@ -88,6 +89,17 @@ export default Mixin.create({
         );
       }
 
+      if (
+        !isEmpty(this.accountName) &&
+        this.accountPassword === this.accountName
+      ) {
+        return EmberObject.create(
+          Object.assign(failedAttrs, {
+            reason: i18n("user.password.same_as_name"),
+          })
+        );
```

Comment:
Should we have a little helper to create "failed" validations so we don't repeat over and over `return EmberObject.create(Object.assign(...))`?

Reply:
I tried, but given the specific use cases of the reason, the helper ends up requiring too many parameters, which results in the same number of lines, if not more